distance_metrics_test.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+from distance_metrics import ImgFormat, Distance
+import numpy as np
+import unittest
+
+class TestVisualGenerator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Temporary image array
+        cls.img_rgb_0 = np.zeros((32, 32, 3), dtype=np.uint8)
+        cls.img_rgb_1 = np.ones((32, 32, 3), dtype=np.uint8)
+        cls.img_gray_0 = np.zeros((32, 32), dtype=np.uint8)
+        cls.img_gray_1 = np.ones((32, 32), dtype=np.uint8)
+        cls.emb_0 = np.zeros(1000, dtype=np.float64)
+        cls.emb_1 = np.ones(1000, dtype=np.float64)
+
+    def test_default_init(self):
+        """Test default initialization. When default initialization value
+        changes, or any private attribute does not match public attribute, this
+        test will fail."""
+        dis = Distance()
+
+        self.assertEqual(dis._img_format, ImgFormat.RGB)
+
+    def test_img_format_setter(self):
+        # Test setter in initialization
+        dis = Distance(img_format=ImgFormat.A8)
+        self.assertEqual(dis._img_format, ImgFormat.A8)
+
+        # Test setter after initialization
+        dis.img_format = ImgFormat.EMBEDDINGS
+        self.assertEqual(dis._img_format, ImgFormat.EMBEDDINGS)
+
+        # Test exception
+        with self.assertRaises(TypeError):
+            dis.img_format = 5
+
+    def test_rgb_metrics(self):
+        # Test RGB metrics
+        dis = Distance(ImgFormat.RGB)
+        metrics = dis.get_metrics()
+        self.assertEqual(metrics['manhattan'](self.img_rgb_0, self.img_rgb_1),
```

Comment:
I am a little bit concern about your test cases here. It seems for each metrics, you only verify two very simple cases. I don't think this is enough to ensure it implement what you intend to implement. Could you create 1 or 2 more images to compare. For example, an image which have one pixel fully on, and another image which have one pixel set to a different value. BTW, why do you need to implement these 4 different image types? Are you using all 4 of these types? Shouldn't we just focusing on a grey scale one instead?

Reply:
Yes. Good idea. Will add more tests. By the time I developed this module, I haven't figured out which of the image format to use, thus the reason for 4 different format. In the end, only 2 formats are used.